marconi-core/test/Marconi/CoreSpec.hs

```diff
@@ -1633,3 +1635,106 @@ withRollbackFailureTest =
   Tasty.testProperty "Rollback failure in a worker exit nicely" $
     Test.withMaxSuccess 10_000 $
       checkOutOfRollback (Lens.view defaultChain <$> Test.arbitrary)
+
+-- * Query modification - Stability
+withStabilityTestGroup :: Tasty.TestTree
+withStabilityTestGroup =
+  Tasty.testGroup
+    "StabilityTests"
+    [ Tasty.testGroup
+        "Stability"
+        [ Tasty.testGroup
+            "stability tests"
+            [ Tasty.testProperty "all results are stable" allStable
+            , Tasty.testProperty "not all results are stable" notAllStable
+            , Tasty.testProperty "withStability will not alter the events" withStabilityUnchanged
+            , Tasty.testProperty "no results are stable at a specific point" notStableAt
+            , Tasty.testProperty "all results are stable at a specific point" stableAt
+            , Tasty.testProperty "withStabilityAt will not alter the events" withStabilityAtUnchanged
+            ]
```

Comment:
can you group in two different groups the test of `withStability` and `withStabilityAt`?